src/sources/models.js

```diff
@@ -172,18 +176,33 @@ class Resource {
       throw new BadRequest(`Resource (e.g. dataset and narrative) paths may not include underscores (_); use slashes (/) instead`);
     }
     /* Forbid '@' characters in pathParts as these will be used to identify the
-     * version descriptor.
+     * version descriptor. Currently the version descriptor is only used for
+     * dataset resources of the core source.
      */
     if (pathParts.some(part => part.includes("@"))) {
-        throw new BadRequest('This resource does not (yet) allow version descriptors and therefore must not contain "@"');
-      }
+      throw new BadRequest(
+        (this.source.name==='core' && this instanceof Dataset) ?
+          'Core dataset paths can only include "@" if it is used to identify a YYYY-MM-DD version' :
+          'This resource does not (yet) allow version descriptors and therefore must not contain "@"'
```

Comment:
Following on from my comment in an earlier commit, this breaks encapsulation boundaries and erodes the ability to reason about bits of the model classes separately.

Reply:
This is now no longer needed (and is removed) - see above commentary about extracting version descriptors for all requests.